### PR TITLE
Add CODEOWNERS for core components

### PR DESCRIPTION
**Pull request by mx-psi**
**Description:** <Describe what has changed.>

Add CODEOWNERs for several shared components. For most of them I added core approvers, some are only contrib.

**Link to tracking Issue:** Updates #3870

### CODE REVIEW

**Comment by mx-psi** [comment]
cc @open-telemetry/collector-contrib-approvers

**Review by TylerHelmuth — APPROVED** [review]

**Review by bogdandrutu — APPROVED** [review]

**Comment by bogdandrutu** [comment]
Please fix the conflicts

File: .github/CODEOWNERS [hunk 1]
```diff
@@ -212,4 +217,6 @@ receiver/windowsperfcountersreceiver/                @open-telemetry/collector-c
 receiver/zipkinreceiver/                             @open-telemetry/collector-contrib-approvers
 receiver/zookeeperreceiver/                          @open-telemetry/collector-contrib-approvers @djaglowski
 
+
+testbed/                                             @open-telemetry/collector-contrib-approvers @open-telemetry/collector-approvers
```
Comment (dmitryax) [inline .github/CODEOWNERS:221]:
```suggestion
testbed/                                             @open-telemetry/collector-contrib-approvers @open-telemetry/collector-approvers
```

**Review by dmitryax — COMMENTED** [review]
nit

**Review by codeboten — APPROVED** [review]

**Pull request merged**